skxray/diff_roi_choice.py

```diff
@@ -375,3 +378,117 @@ def _process_rings(num_rings, img_dim, ring_vals, ring_inds):
     num_pixels = num_pixels[1:]
 
     return ring_inds, ring_vals, num_pixels, pixel_list
+
+
+def roi_divide_circle(detector_size, radius, calibrated_center,
+                      num_angles):
+    """
+    This module will provide the indices, number of pixels and
+    pixel indices when a circular roi is divided into pies
+    Parameters
+    ----------
+    detector_size : tuple
+        shape of the image (detector X and Y direction)
+        Order is (num_rows, num_columns)
+
+    radius : float
+        radius of the circle
+
+    calibrated_center : tuple
+        defining the center of the image
+        (column value, row value) (mm)
+
+    num_angles: int
+        number of angles ring divide into
+        angles are measured from horizontal-anti clock wise
+
+    Returns
+    -------
+    ring_inds : ndarray
+        indices of the ring values for the required rings
+        (after discarding zero values from the shape
+        ([detector_size[0]*detector_size[1]], )
+
+    num_pixels : ndarray
+        number of pixels in each ring
+
+    pixel_list : ndarray
+        pixel indices for the required rings
+
+    """
+    mesh = np.zeros(detector_size, dtype=np.int64)
+    angles = np.linspace(0, 360, num_angles)
+
+    yy, xx = np.mgrid[:detector_size[0], :detector_size[1]]
+    y_ = (np.flipud(yy) - calibrated_center[1])
+    x_ = (xx - calibrated_center[0])
+    grid_values = np.float_(np.hypot(x_, y_))
+    angle_grid = np.arctan2(y_, x_)*180/(np.pi)
+
+    angle_grid[grid_values > radius]=np.nan
+
+    for i in range(num_angles-1):
+        if angles[i + 1] <= 180:
+            vl = (angles[i] < angle_grid) & (angle_grid <= angles[i + 1])
+
+        elif (angles[i] < 180) & (angles[i + 1] > 180):
+            vl = ((angles[i] < angle_grid) & (angle_grid <= 180) |
+                  (-180 < angle_grid) & (angle_grid <= angles[ i + 1] - 360))
+
+        elif (angles[i]==180):
+            vl= (-180 < angle_grid) & (angle_grid <= angles[i + 1] - 360)
+
+        else:
+            vl = (angles[i] - 360 < angle_grid) & (angle_grid <= angles[i + 1]- 360)
+
+        mesh[vl] = i + 1
+
+    roi_inds, num_pixels, pixel_list = _process_rois(np.ravel(mesh),
+                                                     detector_size,
+                                                     num_angles)
+    return roi_inds, num_pixels, pixel_list
+
+
+def _process_rois(mesh, detector_size, num_rois):
```

Comment:
Yes _process_rings() changed it, now it will only process rings edge values. 
This module will take care of all the modules final processing of finding pixel_list, number of pixels, indices